sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5186,8 +5335,8 @@ pi_result _pi_event::cleanup(pi_queue LockedQueue) {
     // Lock automatically releases when this goes out of scope.
     auto Lock = ((Queue == LockedQueue) ? std::unique_lock<pi_shared_mutex>()
                                         : std::unique_lock(Queue->Mutex));
-
-    if (ZeCommandList) {
+    // Immediate commandlists do not have a fence, so skip this step
```

Comment:
I think it would be more accurate to say that we don't enter immediate command-lists to the `CommandListMap`, right? BTW, the event's `ZeCommandList` still points to a valid immediate L0 command-list, correct?

Reply:
We _do_ enter immediate commandlists in the map. That's where we can locate all of them to destroy them at queue deletion. However, the rest of the body of that if-stmt has to do with the fence, which imm cmdlists lack. Removing the UseImmediateCommandLists check will be OK.

Reply:
if (it == Queue->CommandListMap.end()) {
          die("Missing command-list completition fence");
is not exactly intuitive. It is existing code.